src/audio/module_adapter/module_adapter.c

```diff
@@ -626,54 +633,124 @@ static void module_adapter_process_output(struct comp_dev *dev)
 	}
 }
 
+static uint32_t
+module_single_sink_setup(struct comp_dev *dev,
+			 struct comp_buffer __sparse_cache *source_c[PLATFORM_MAX_STREAMS],
+			 struct comp_buffer __sparse_cache *sinks_c[PLATFORM_MAX_STREAMS])
```

Comment:
@lyakh I think this is to be compatible with the module API, i.e. the call to process will pass an array of inputs/outputs.
@ranj063 doesn't the module API use variable size arrays instead of fixed size (as here). I think it's best to align.

Reply:
Sure I can make these pointers